### ISSUE TITLE
Build issue: AutoUpdater assembly is not signed with a strong name

### ISSUE THREAD

**Issue by warwickmm** [duplicati/duplicati#2814]
As of revision 762035c309, building the solution using `mcs` results in a build error:
```
error CS1577: Referenced assembly `Duplicati.Library.AutoUpdater, Version=2.0.0.7, Culture=neutral, PublicKeyToken=null' does not have a strong name
```
When using `xbuild`, this results in a warning:
```
CSC:  warning CS8002: Referenced assembly 'Duplicati.Library.AutoUpdater, Version=2.0.0.7, Culture=neutral, PublicKeyToken=null' does not have a strong name.
```
Should this assembly be signed to prevent potential tampering?

**Comment by kenkendk** [comment]
I have actually started removing the strong-names, as they cause a lot of troubles for open-source projects. According to (L)GPL licenses, the end user must be able to replace the library/program with a newer version, which can only happen if I share the key.

Sharing the key means that the SN is really misleading as it does not really add the security it should. It would protect against casual tampering, but not deliberate tampering.

Because of this (or because of ignorance?) many of the libraries that Duplicati depends on are not signed. Previously I used `signer.exe` and manually signed the assemblies, but this approach is not compatible with NuGet, so I stopped doing.

I would appreciate anyone sharing opinons, thoughts or plans for SN support in Duplicati :)

**Comment by warwickmm** [comment]
The subtleties of the license definitely lead to interesting conundrums.  I think you're right in that removing strong names is probably the simplest way to go.

It looks like some projects (e.g., [Json.NET](http://james.newtonking.com/archive/2012/04/04/json-net-strong-naming-and-assembly-version-numbers)) use the same assembly version for minor (or all) releases.  However, this breaks the convention that [assemblies that have the same strong name should be identical](https://docs.microsoft.com/en-us/dotnet/framework/app-domains/strong-named-assemblies), and can lead to [other issues](https://github.com/JamesNK/Newtonsoft.Json/issues/615) as well.

The issue mentioned in the first post is due to the `WindowsService` project (which is signed) referencing the `AutoUpdater` project (which is not signed).  While some compilers (`xbuild`, `roslyn`) appear to generate only warnings, this can lead to [runtime errors](https://github.com/dotnet/roslyn/issues/7277) depending on the CLR.  

Would it make sense to stop signing at least the `WindowsService` project, if not all of the projects?  I count 26 projects with `<SignAssembly>true</SignAssembly>` in their `.csproj` files.

**Comment by kenkendk** [comment]
Yes, for now I think the easy fix is to not sign the top-level projects (including `WindowsService`).
But maybe we should just remove all use of `<SignAssembly>`.

**Comment by kenkendk** [comment]
Fixed with 6375211ec907d63ba61a071bf70279751572015c
